tardis/plasma/properties/nlte_rate_equation_solver.py

```diff
@@ -666,29 +667,29 @@ def population_objective_function(
             jacobian_matrix,
         )
 
-    def solution_vector_block(self, atomic_number, number_density):
+    def solution_vector_block(self, number_density, needed_number_of_elements):
         """Block of the solution vector for the current atomic number.
 
         Block for the solution vector has the form (0, 0, ..., 0, number_density).
         Length is equal to atomic_number+1.
 
         Parameters
         ----------
-        atomic_number : int
-            Current atomic number.
         number_density : float
             Number density of the current atomic number.
+        needed_number_of_elements : int
+            Number of needed elements for current atomic number.
```

Comment:
Maybe add "... elements in the solution vector ..." to be more specific? I was a bit confused initially because I was thinking of chemical elements.